player/configfiles.c

```diff
@@ -237,60 +237,55 @@ static char *mp_get_playback_resume_config_filename(struct MPContext *mpctx,
     return res;
 }
 
-static const char *const backup_properties[] = {
-    "osd-level",
-    //"loop",
-    "speed",
-    "options/edition",
-    "pause",
-    "volume",
-    "mute",
-    "audio-delay",
-    //"balance",
-    "fullscreen",
-    "ontop",
-    "border",
-    "gamma",
-    "brightness",
-    "contrast",
-    "saturation",
-    "hue",
-    "options/deinterlace",
-    "vf",
-    "af",
-    "panscan",
-    "options/aid",
-    "options/vid",
-    "options/sid",
-    "sub-delay",
-    "sub-speed",
-    "sub-pos",
-    "sub-visibility",
-    "sub-scale",
-    "sub-use-margins",
-    "sub-ass-force-margins",
-    "sub-ass-vsfilter-aspect-compat",
-    "sub-ass-override",
-    "ab-loop-a",
-    "ab-loop-b",
-    "options/video-aspect-override",
-    0
-};
-
 // Used to retrieve default settings, which should not be stored in the
-// resume config. Uses backup_properties[] meaning/order of values.
+// resume config. Uses watch_later_properties[] meaning/order of values.
 // This explicitly includes values set by config files and command line.
 void mp_get_resume_defaults(struct MPContext *mpctx)
 {
-    char **list =
-        talloc_zero_array(mpctx, char*, MP_ARRAY_SIZE(backup_properties));
-    for (int i = 0; backup_properties[i]; i++) {
-        const char *pname = backup_properties[i];
-        char *val = NULL;
-        int r = mp_property_do(pname, M_PROPERTY_GET_STRING, &val, mpctx);
-        if (r == M_PROPERTY_OK)
-            list[i] = talloc_steal(list, val);
+    char **watch_later_properties = mpctx->opts->watch_later_properties;
+
+    // --watch-later-properties= makes the first element "",
+    // while change-list watch-later-properties clr "" frees the list.
+    if (!watch_later_properties ||
+        (strcmp(watch_later_properties[0], "") == 0 && !watch_later_properties[1])
+       ) {
+        talloc_free(mpctx->resume_defaults);
```

Comment:
According to `MPV_LEAK_REPORT`, this actually doesn't leak so this `talloc_free` (nor the other one) are needed. I think that makes sense because `mpctx` is the parent which, of course, already has its own memory allocated to it. `MP_TARRAY_APPEND` appears to only increase the allocated size when necessary, so I think this all memory stuff just works(tm).

Reply:
If you `change-list watch-later-properties` 1000 times the `resume_defaults` are freed on exit, but wouldn't the memory keep accumulating until then?

Reply:
Yup you're right.